src/test/java/org/prebid/server/handler/ForceDealsUpdateHandlerTest.java

```diff
@@ -0,0 +1,243 @@
+package org.prebid.server.handler;
+
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.prebid.server.deals.AlertHttpService;
+import org.prebid.server.deals.DeliveryProgressService;
+import org.prebid.server.deals.DeliveryStatsService;
+import org.prebid.server.deals.LineItemService;
+import org.prebid.server.deals.PlannerService;
+import org.prebid.server.deals.RegisterService;
+import org.prebid.server.exception.PreBidException;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+
+public class ForceDealsUpdateHandlerTest {
+
+    private static final String ACTION_NAME_PARAM = "action_name";
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private DeliveryStatsService deliveryStatsService;
+    @Mock
+    private PlannerService plannerService;
+    @Mock
+    private RegisterService registerService;
+    @Mock
+    private AlertHttpService alertHttpService;
+    @Mock
+    private DeliveryProgressService deliveryProgressService;
+    @Mock
+    private LineItemService lineItemService;
+
+    private ForceDealsUpdateHandler handler;
+
+    @Mock
+    private RoutingContext routingContext;
+    @Mock
+    private HttpServerRequest httpRequest;
+    @Mock
+    private HttpServerResponse httpResponse;
+
+    @Before
+    public void setUp() {
+        handler = new ForceDealsUpdateHandler(
+                deliveryStatsService,
+                plannerService,
+                registerService,
+                alertHttpService,
+                deliveryProgressService,
+                lineItemService,
+                "/endpoint");
+
+        given(routingContext.request()).willReturn(httpRequest);
+        given(routingContext.response()).willReturn(httpResponse);
+
+        given(httpResponse.setStatusCode(anyInt())).willReturn(httpResponse);
+        given(httpResponse.closed()).willReturn(false);
+    }
+
+    @Test
+    public void shouldReturnBadRequestWhenActionParamIsMissing() {
+        // given
+        given(httpRequest.getParam(any())).willReturn(null);
+
+        // when
+        handler.handle(routingContext);
+
+        //  then
+        verify(httpRequest).getParam(eq(ACTION_NAME_PARAM));
+
+        verify(httpResponse).setStatusCode(eq(400));
+        verify(httpResponse).end(eq(String.format("Parameter '%s' is required and can't be empty", ACTION_NAME_PARAM)));
```

Comment:
Minor. eq() can be omitted. `verify(httpResponse).end(String.format("Parameter '%s' is required and can't be empty", ACTION_NAME_PARAM))`